File: Lab3/src/KI/Kulyk/Lab3/MobilePhone.java
package KI.Kulyk.Lab3;

/**
 * Інтерфейс з функціоналом для мобільного телефону.
 */
interface MobileDevice {
    void sendSMS(String number, String message);
    void receiveSMS(String message);
}

/**
 * Клас MobilePhone розширює Telephone і реалізує інтерфейс MobileDevice.
 */
public class MobilePhone extends Telephone implements MobileDevice {

    private String phoneNumber;

    /**
     * Конструктор з параметрами.
     *
     * @param display   дисплей телефону
     * @param battery   батарея телефону
     * @param processor процесор телефону
     * @param phoneNumber номер телефону
     */
    public MobilePhone(Display display, Battery battery, Processor processor, String phoneNumber) {
        super(display, battery, processor);
        this.phoneNumber = phoneNumber;
    }

    @Override
    public void makeCall(String number) {
        setBatteryCapacity(getBatteryCapacity() - 30);
        log("Calling to " + number + " from " + phoneNumber + ".");
    }

    @Override
    public void sendSMS(String number, String message) {
        setBatteryCapacity(getBatteryCapacity() - 15);
        log("Sending SMS to " + number + ": " + message);
    }

    @Override
    public void receiveSMS(String message) {
        setBatteryCapacity(getBatteryCapacity() - 5);
        log("Received SMS: " + message);
    }
}

File: MobilePhone.py
from Phone import Phone

class MobilePhone(Phone):
    def __init__(self, display, battery, processor, phone_number):
        super().__init__(display, battery, processor)
        self.phone_number = phone_number

    def make_call(self, number):
        self.set_battery_capacity(self.get_battery_capacity() - 30)
        print(f"Calling to {number} from {self.phone_number}.")

    def send_sms(self, number, message):
        self.set_battery_capacity(self.get_battery_capacity() - 15)
        print(f"Sending SMS to {number}: {message}")

    def receive_sms(self, message):
        self.set_battery_capacity(self.get_battery_capacity() - 5)
        print(f"Received SMS: {message}")
